ui/apps/platform/src/hooks/useURLSort.ts

```diff
@@ -45,65 +34,161 @@ function tableSortOption(
+ *     - This signature is used when the column has a single sort field. The `columnName` parameter
+ *     is used to find the correct column index in the PatternFly table, and is also used as the default
+ *     field name when sending the sort option to the API. The `fieldOptions` parameter is optional and
+ *     is used to specify the aggregate function to use when sorting the column.
+ *
+ *     2. getSortParams(columnName: string, fieldOptions: { field: string; aggregateBy?: SortAggregate }[])
+ *     - This signature is used when the column has multiple sort fields. In this form, the fields sent to
+ *     the API will be the `field` values in the `fieldOptions` array, each of which can optionally
+ *     include an aggregate. The correct column index in the PatternFly table is found by finding any array
+ *     provided in the `sortFields` option that contains one of the `field` values in the `fieldOptions` array.
+ *
+ * @example
+ * // A use case where a table only has single sort columns
+ * const { sortOption, setSortOption, getSortParams } = useURLSort({
+ *    sortFields: ['name', 'age', 'date'],
+ *    defaultSortOption: { field: 'name', direction: 'asc' },
+ * });
+ *
+ * return (
+ *   <Thead>
+ *     <Th sort={getSortParams('name')}>Name</Th>
+ *     <Th sort={getSortParams('age')}>Age</Th>
+ *     <Th sort={getSortParams('date')}>Date</Th>
+ *   </Thead>
+ * );
+ *
+ * @example
+ * // A use case where a table has both single and multi sort columns
+ * const { sortOption, setSortOption, getSortParams } = useURLSort({
+ *   sortFields: ['name', 'age', 'date', ['critical count', 'important count']],
+ *  defaultSortOption: { field: 'name', direction: 'asc' },
+ * });
+ *
+ * return (
+ *   <Thead>
+ *     <Th sort={getSortParams('name')}>Name</Th>
+ *     <Th sort={getSortParams('age')}>Age</Th>
+ *     <Th sort={getSortParams('date')}>Date</Th>
+ *    <Th sort={getSortParams('critical count', [
+ *       { field: 'critical count' },
+ *       { field: 'important count' },
+ *    ])}>Critical Count</Th>
+ *   </Thead>
+ * );
+ *
+ *
+ */
 function useURLSort({ sortFields, defaultSortOption, onSort }: UseURLSortProps): UseURLSortResult {
     const [sortOption, setSortOption] = useURLParameter('sortOption', defaultSortOption);
 
     // get the parsed sort option values from the URL, if available
     // otherwise, use the default sort option values
-    const activeSortField =
-        isParsedQs(sortOption) && typeof sortOption?.field === 'string'
-            ? sortOption.field
-            : defaultSortOption.field;
-    const activeSortDirection =
-        isParsedQs(sortOption) && isDirection(sortOption?.direction)
-            ? sortOption.direction
-            : defaultSortOption.direction;
-    const activeAggregateBy =
-        isParsedQs(sortOption) && sortOption?.aggregateBy
-            ? (sortOption?.aggregateBy as SortAggregate)
-            : undefined;
```

Comment:
This already complex logic would be more complex with multi-sort option support, so was moved to a helper function in `getActiveSortOption`. Additionally, `isParsedQs` was replaced with a yup validator to ensure the URL value conforms to the correct type.